bundles/org.openhab.binding.fineoffsetweatherstation/src/main/java/org/openhab/binding/fineoffsetweatherstation/internal/domain/Measurands.java

```diff
@@ -0,0 +1,171 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.fineoffsetweatherstation.internal.domain;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.fineoffsetweatherstation.internal.FineOffsetWeatherStationBindingConstants;
+import org.openhab.binding.fineoffsetweatherstation.internal.domain.response.MeasuredValue;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.State;
+
+/**
+ * Holds all the measurands supported by the gateway.
+ *
+ * @author Andreas Berger - Initial contribution
+ */
+@NonNullByDefault
+public class Measurands {
+
+    private static final Map<Protocol, Measurands> INSTANCES = new HashMap<>();
+    private final Map<Byte, List<Parser>> parsersPerCode = new HashMap<>();
+
+    private Measurands(Protocol protocol) {
+        try (InputStream data = Measurands.class.getResourceAsStream("/measurands.csv")) {
```

Comment:
Just for my understanding: What is the benefit of parsing this CSV resource compared to previous coded solution? Second, if this approach is beneficial, would it be worth considering JSON and getting rid of **commons-csv**, or would it be too bloated?

Reply:
> What is the benefit of parsing this CSV resource compared to previous coded solution?
The CSV is clearer, and the differences between the individual protocols are easier to maintain.

If this were mapped in Java, the constructor would become even more complex than it already was.
> Second, if this approach is beneficial, would it be worth considering JSON and getting rid of commons-csv, or would it be too bloated?

Is there any problem with common-csv? The possibility of editing would not be as easy with JSON as it is with CSV.

If CSV is a problem, I would then take the Java approach again and put everything into code.

Reply:
I was remembering #7722, so I believe it should be avoided. Nothing wrong with CSV though, perhaps you could even parse this in a simple way with custom code?

Reply:
> perhaps you could even parse this in a simple way with custom code?

Sure but why should I implement a CSV-Parser, if I can use a rock solid one?

So should I remove the dependency? If so, I would turn the data back into code...

Reply:
Looking at the code, you wouldn't need to implement a full-blown CSV parser, just some simple line-parser using a separator. Before you revert anything, let's hear @openhab/add-ons-maintainers - is it correctly understood, according to #7722, that usage of org.apache.common.* is no longer allowed, including org.apache.commons.csv?

Reply:
> Something like this could be considered: https://www.baeldung.com/java-csv-file-array#buff-reader

Yes, I know, but this solution does not support column-based lookup, which is what I use.

Reply:
@cweitkamp, @kaikreuzer - approaching you directly, since I'm guessing you would know best: Should we avoid usage of `org.apache.common.*` (including `org.apache.commons.csv` as used here), as mentioned in #7722? I would not like to reintroduce usage after a previous cleanup effort, but may have misunderstood intentions, so would like confirmation before blocking this PR with such requirement.

Reply:
The general idea behind the removal of dependencies is to avoid bloated code where not necessary. For me it is okay if a binding still uses `org.apache.commons.*` for special purposes if the feature is needed and no different solution can be found. e.g. the mail binding uses `commons-email`, dbquery and influx are using `commons-csv`. Maybe the versions can be aligned.

Reply:
@cweitkamp - thanks for adding some context. The two mentioned bundles use version 1.6, but we can try to upgrade them to 1.9 for alignment. @Andy2003, I'm resolving this comment since you now have a go-ahead to use **commons-csv**.